src/interfaces.ts

```diff
@@ -16,7 +16,31 @@ export interface FileExplorerItem {
 
 export interface TagExplorerItem {
   tag: string;
-  children: TagExplorerItem[];
+  children?: TagExplorerItem[];
   collapsed?: boolean;
   setCollapsed?: (state: boolean) => void;
+  vChildren?: {
+    _children: TagExplorerItem[]
+  };
+}
+
+export class TagExplorerItemImpl implements TagExplorerItem {
+  constructor(item: TagExplorerItem) {
+    Object.assign(this, item);
+    this.setCollapsed = item.setCollapsed.bind(item);
+  }
+
+  tag: string;
+  children?: TagExplorerItem[];
+  collapsed?: boolean;
+  setCollapsed?: (state: boolean) => void;
+  vChildren?: { _children: TagExplorerItem[]; };
+
+  getChildrenSafe(): TagExplorerItem[] {
```

Comment:
Let's just move this logic to the `TagPaneProvider` as a private method, then you can just replace any instances of `.children` with a call to the method.